File: NEsper.Tests/support/util/SerializableObjectCopier.cs
// ---------------------------------------------------------------------------------- /
// Copyright (C) 2007 Esper Team. All rights reserved.                                /
// http://esper.codehaus.org                                                          /
// ---------------------------------------------------------------------------------- /
// The software in this package is published under the terms of the GPL license       /
// a copy of which has been included with this distribution in the license.txt file.  /
// ---------------------------------------------------------------------------------- /

using System;
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Formatters.Binary;

namespace net.esper.support.util
{
	public class SerializableObjectCopier
	{
	    public static Object Copy(Object orig)
	    {
	        BinaryFormatter formatter = new BinaryFormatter();

            MemoryStream stream = new MemoryStream();
            formatter.Serialize(stream, orig);
            stream.Flush();

	        stream.Seek(0, SeekOrigin.Begin);
	        Object result = formatter.Deserialize(stream);

	        return result;
	    }
	}
} // End of namespace

File: trunk/esper/src/test/java/com/espertech/esper/support/util/SerializableObjectCopier.java
package com.espertech.esper.support.util;

import java.io.ObjectOutputStream;
import java.io.ObjectInputStream;
import java.io.IOException;

public class SerializableObjectCopier
{
    public static Object copy(Object orig) throws IOException, ClassNotFoundException
    {
        SimpleByteArrayOutputStream fbos = new SimpleByteArrayOutputStream();
        ObjectOutputStream out = new ObjectOutputStream(fbos);
        out.writeObject(orig);
        out.flush();
        out.close();

        ObjectInputStream in = new ObjectInputStream(fbos.getInputStream());
        Object result = in.readObject();
        return result;
    }
}
